PWM Example compile issues

Hello! I'm a beginner with rust and am currently working with a STM32L011K4T6.
I wanted to try out the PWM Example for my project, however, I'm running into some issues that I'm not sure how to fix.

I've copied the example into my project and changed the PWM pin to PB3 and use the feature stm32l0x2.

When compiling, I get the following error messages:

```
user@MacBook-Air l011-blinky % cargo build --release
   Compiling l011-blinky v0.1.0 (/Users/user/Documents/Projekte/l011-blinky)
error[E0599]: no method named `get_max_duty` found for struct `stm32l0xx_hal::pwm::Timer` in the current scope
  --> src/main.rs:33:19
   |
33 |     let max = pwm.get_max_duty();
   |                   ^^^^^^^^^^^^ method not found in `stm32l0xx_hal::pwm::Timer<stm32l0xx_hal::pac::TIM2>`

error[E0599]: no method named `enable` found for struct `stm32l0xx_hal::pwm::Timer` in the current scope
  --> src/main.rs:35:9
   |
35 |     pwm.enable();
   |         ^^^^^^ method not found in `stm32l0xx_hal::pwm::Timer<stm32l0xx_hal::pac::TIM2>`

error[E0599]: no method named `set_duty` found for struct `stm32l0xx_hal::pwm::Timer` in the current scope
  --> src/main.rs:38:13
   |
38 |         pwm.set_duty(max);
   |             ^^^^^^^^ method not found in `stm32l0xx_hal::pwm::Timer<stm32l0xx_hal::pac::TIM2>`

error[E0599]: no method named `set_duty` found for struct `stm32l0xx_hal::pwm::Timer` in the current scope
  --> src/main.rs:41:13
   |
41 |         pwm.set_duty(max / 2);
   |             ^^^^^^^^ method not found in `stm32l0xx_hal::pwm::Timer<stm32l0xx_hal::pac::TIM2>`

error[E0599]: no method named `set_duty` found for struct `stm32l0xx_hal::pwm::Timer` in the current scope
  --> src/main.rs:44:13
   |
44 |         pwm.set_duty(max / 4);
   |             ^^^^^^^^ method not found in `stm32l0xx_hal::pwm::Timer<stm32l0xx_hal::pac::TIM2>`

error[E0599]: no method named `set_duty` found for struct `stm32l0xx_hal::pwm::Timer` in the current scope
  --> src/main.rs:47:13
   |
47 |         pwm.set_duty(max / 8);
   |             ^^^^^^^^ method not found in `stm32l0xx_hal::pwm::Timer<stm32l0xx_hal::pac::TIM2>`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `l011-blinky` due to 6 previous errors
```

I've checked the documentation, and my understanding is that the required features are implemented at the [PWM](https://docs.rs/stm32l0xx-hal/0.10.0/stm32l0xx_hal/pwm/struct.Pwm.html#method.get_max_duty) struct and not the [Timer](https://docs.rs/stm32l0xx-hal/0.10.0/stm32l0xx_hal/pwm/struct.Timer.html#structfield.channel2) struct. However, due to my limited experience in Rust I'm not sure how to use instantiate the PWM peripheral correctly. Would it be possible to update the example?

Thank you very much :)
I've tried your PR, but sadly I still get the same result as before. My Cargo.toml is looking like this:

```
[package]
name = "l011_blinky"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
cortex-m = "0.7.6"
cortex-m-rt = "0.7.1"
embedded-hal = "0.2.7"
panic-halt = "^0.2.0"
stm32l0 = "0.15.1"
stm32l0xx-hal = {git = "https://github.com/stm32-rs/stm32l0xx-hal.git", branch = "correct-tim2-pins", features = ["mcu-STM32L011K4Tx", "rt"]}
```

I've also tried using TIM21 instead of TIM2 for PWM, but it seems like this is not implemented?
I get the additional error message:

```
error[E0277]: the trait bound `stm32l0xx_hal::pac::TIM21: stm32l0xx_hal::pwm::Instance` is not satisfied
  --> src/main.rs:27:31
   |
27 |     let pwm = pwm::Timer::new(dp.TIM21, 10_000.Hz(), &mut rcc);
   |               --------------- ^^^^^^^^ the trait `stm32l0xx_hal::pwm::Instance` is not implemented for `stm32l0xx_hal::pac::TIM21`
   |               |
   |               required by a bound introduced by this call
   |
   = help: the following other types implement trait `stm32l0xx_hal::pwm::Instance`:
             stm32l0xx_hal::pac::TIM2
             stm32l0xx_hal::pac::TIM3

```

Is it not intended to use TIM21 for PWM? What about the Low Power Timers?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,10 +11,11 @@ make upgrading the HAL as painless as possible! If it makes sense, feel free
 to add upgrade notes and examples. When adding an issue or PR reference, don't
 forget to update the links at the bottom of the changelog as well.-->
 
-- Enable SPI2 on subset of stm32l0x1 devices ([#221])
-
 ### Additions
 
+- Enable TIM2 outputs on `PA5`, `PA15`, `PB3` for all devices in the L0 family (previously only 0x2
+  and 0x3) ([#224])
+- Enable SPI2 on subset of stm32l0x1 devices ([#221])
 - Add `pause` and `resume` methods to timers ([#220])
 
 ### Breaking Changes
@@ -185,6 +186,8 @@ _Not yet tracked in this changelog._
 
 <!-- Links to pull requests and issues. -->
 
+[#224]: https://github.com/stm32-rs/stm32l0xx-hal/pull/224
+[#221]: https://github.com/stm32-rs/stm32l0xx-hal/pull/221
 [#220]: https://github.com/stm32-rs/stm32l0xx-hal/pull/218
 [#218]: https://github.com/stm32-rs/stm32l0xx-hal/pull/218
 [#215]: https://github.com/stm32-rs/stm32l0xx-hal/pull/215

--- a/src/pwm.rs
+++ b/src/pwm.rs
@@ -1,4 +1,8 @@
 use crate::gpio::gpioa::{PA0, PA1, PA2, PA3};
+use crate::gpio::{
+    gpioa::{PA15, PA5},
+    gpiob::PB3,
+};
 use crate::gpio::{AltMode, PinMode};
 use crate::hal;
 use crate::pac::{tim2, TIM2, TIM3};
@@ -10,10 +14,7 @@ use cortex_m::interrupt;
 use embedded_time::rate::Hertz;
 
 #[cfg(any(feature = "stm32l0x2", feature = "stm32l0x3"))]
-use crate::gpio::{
-    gpioa::{PA15, PA5},
-    gpiob::{PB10, PB11, PB3},
-};
+use crate::gpio::gpiob::{PB10, PB11};
 
 #[cfg(any(feature = "stm32l072", feature = "stm32l082", feature = "io-STM32L071",))]
 use crate::gpio::{
@@ -299,15 +300,15 @@ impl_pin!(
         PA1, C2, AF2;
         PA2, C3, AF2;
         PA3, C4, AF2;
+        PA5,  C1, AF5;
+        PA15, C1, AF5;
+        PB3,  C2, AF2;
     )
 );
 
 #[cfg(any(feature = "stm32l0x2", feature = "stm32l0x3"))]
 impl_pin!(
     TIM2: (
-        PA5,  C1, AF5;
-        PA15, C1, AF5;
-        PB3,  C2, AF2;
         PB10, C3, AF2;
         PB11, C4, AF2;
     )

